docu/SICKAppSpaceCodingStarterKit_Documentation.md

```diff
@@ -667,6 +713,451 @@ Script.serveFunction("CSK_ModuleName.setSelectedInstance", setSelectedInstance)
+
+First of all, create a "FlowConfig" folder at 'scripts/Mainfolder/Subfolder/' and create a file '[ModuleName]_FlowConfig.lua' in it.
+
+This will be required out of the main file and internally requires all lua files for available FlowConfig features. Additionally it will register to the 'OnClearOldFlow'-event of the FlowConfig CSK module to optionally clear all FlowConfig relevant configuration if a new flow was activated:
+
+```
+require('Communication.FTPClient.FlowConfig.FTPClient_Put')
+--require('Communication.FTPClient.FlowConfig.FTPClient_OtherFeatureA')
+--require('Communication.FTPClient.FlowConfig.FTPClient_OtherFeatureB')
+
+--- Function to react if FlowConfig was updated
+local function handleOnClearOldFlow()
+  if _G.availableAPIs.specific then
+    -- for i = 1, #multiModuleName_Instances do -- only relevant for MultiModules
+      if ftpClient_Model.parameters.flowConfigPriority then
+        CSK_FTPClient.clearFlowConfigRelevantConfiguration()
+      end
+	-- end
+  end
+end
+Script.register('CSK_FlowConfig.OnClearOldFlow', handleOnClearOldFlow)
+
+--[[
+Only relevant for MultiModules
+local function setMultiModuleName_Instances_Handle(handle)
+  multiModuleName_Instances = handle
+end
+
+return setMultiModuleName_Instances_Handle
+]]
+```
+
+The code content of the FlowConfig files depends if the module should only provide data as an output, only consume incoming data or do both of it.
+
+#### Pure Data Provider
+
+Create a file like 'ModuleName_OnNewData.lua' within the 'FlowConfig' folder and require this within the 'Mainfolder/Subfolder/[ModuleName]_FlowConfig.lua' as mentioned above.
+
+This file should include code like this:
+
+```
+-- Block namespace
+local BLOCK_NAMESPACE = "ModuleName_FC.OnNewData"
+local nameOfModule = 'CSK_ModuleName'
+
+--*************************************************************
+--*************************************************************
+
+local function register(handle, _ , callback)
+
+  Container.remove(handle, "CB_Function")
+  Container.add(handle, "CB_Function", callback)
+
+  local function localCallback()
+    if callback ~= nil then
+      Script.callFunction(callback, 'CSK_ModuleName.OnNewData') -- adapt this accordingly
+    else
+      _G.logger:warning(nameOfModule .. ": " .. BLOCK_NAMESPACE .. ".CB_Function missing!")
+    end
+  end
+  Script.register('CSK_FlowConfig.OnNewFlowConfig', localCallback)
+
+  return true
+end
+Script.serveFunction(BLOCK_NAMESPACE ..".register", register)
+
+--*************************************************************
+--*************************************************************
+
+local function create()
+  local container = Container.create()
+  Container.add(container, "CB_Function", "")
+  return(container)
+end
+Script.serveFunction(BLOCK_NAMESPACE .. ".create", create)
+```
+
+The 'create' function will be called, as soon as a block of this feature exists within the flow.
```

Comment:
Comments:
Text is OK like this, especially if the style it is consistent with other CSK documentation. If this is not too much work, it would be even better if... (optional!)
1) in official programming tutorials we only use quotation marks like "...". For functions we use Camel case just like in the API docs so in this case it would be ... The create() function will be called ...
2) It would be easier to read without "will", but works this way. example: The create() function is called as soon as ...

Reply:
Future issue